packages/lambda-powertools-middleware-log-timeout/index.js

```diff
@@ -0,0 +1,29 @@
+const Log = require('@perform/lambda-powertools-logger')
+
+module.exports = (thresholdMillis = 10) => {
+  return {
+    before: (handler, next) => {
+      const timeLeft = handler.context.getRemainingTimeInMillis()
+      const timeoutMs = timeLeft - thresholdMillis
+      const timer = setTimeout(() => {
+        const awsRequestId = handler.context.awsRequestId
+        const invocationEvent = JSON.stringify(handler.event)
+        Log.error('invocation timed out', { awsRequestId, invocationEvent })
+      }, timeoutMs)
+
+      handler.context.lambdaPowertoolsLogTimeoutMiddleware = {
```

Comment:
should we make this non-enumerable for safety?
```suggestion
      Object.defineProperty(handler.context, 'lambdaPowertoolsLogTimeoutMiddleware', {
        enumerable: false,
        value: { timer }
      });
```

Reply:
yup, good call